tests/sentry/api/endpoints/test_project_team_details.py

```diff
@@ -106,3 +137,45 @@ def test_remove_team_not_found(self):
             "not-a-team",
             status_code=status.HTTP_404_NOT_FOUND,
         )
+
+    @with_feature("organizations:team-roles")
+    def test_remove_team_with_team_role(self):
+        user = self.create_user(username="foo")
+        team_to_remove = self.create_team(organization=self.organization)
+        team_1 = self.create_team(organization=self.organization, slug="admin-team")
+        team_2 = self.create_team(organization=self.organization, slug="contri-team")
+        project_1 = self.create_project(
+            organization=self.organization, teams=[team_1, team_to_remove]
+        )
+        project_2 = self.create_project(
+            organization=self.organization, teams=[team_2, team_to_remove]
+        )
+
+        self.create_member(user=user, organization=self.organization, role="member")
+        self.create_team_membership(user=user, team=team_1, role="admin")
+        self.create_team_membership(user=user, team=team_2)
+        self.login_as(user=user)
+
+        # Cannot revoke access for other team if you are not their admin
```

Comment:
so you can grant project access to another team if you are any sort of team admin, but can only remove if you are also the other team's team admin?

Reply:
Yes, correct too.